Bottom: 7f7b7927438406775fd2a84ed7bc9d203b5f54f9
Top:    c2ab742d6e0dbdc1bdde4a509380245832062b5f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-05 00:21:46 +0100

Refresh of more-play-with-quad-edge-data

---

diff --git a/src/hid/gtk/appearance.c b/src/hid/gtk/appearance.c
new file mode 100644
index 0000000..92790fcc
--- /dev/null
+++ b/src/hid/gtk/appearance.c
@@ -0,0 +1,25 @@
+#include <glib.h>
+
+#include "appearance.h"
+
+appearance
+*make_appearance (void)
+{
+  appearance *appear = g_new0 (appearance, 1);
+
+  return appear;
+}
+
+void
+destroy_appearance (appearance *appear)
+{
+  g_free (appear);
+}
+
+void
+appearance_set_color (appearance *appear, float r, float g, float b)
+{
+  appear->r = r;
+  appear->g = g;
+  appear->b = b;
+}
diff --git a/src/hid/gtk/appearance.h b/src/hid/gtk/appearance.h
new file mode 100644
index 0000000..d4917d7
--- /dev/null
+++ b/src/hid/gtk/appearance.h
@@ -0,0 +1,7 @@
+typedef struct {
+  float r, g, b;
+} appearance;
+
+appearance *make_appearance (void);
+void destroy_appearance (appearance *appearance);
+void appearance_set_color (appearance *appearance, float r, float g, float b);
diff --git a/src/hid/gtk/contour3d.c b/src/hid/gtk/contour3d.c
new file mode 100644
index 0000000..6aa3189
--- /dev/null
+++ b/src/hid/gtk/contour3d.c
@@ -0,0 +1,22 @@
+#include <glib.h>
+#include <stdbool.h>
+
+#include "quad.h"
+#include "contour3d.h"
+
+contour3d *
+make_contour3d (edge_ref first_edge)
+{
+  contour3d *contour;
+
+  contour = g_new0 (contour3d, 1);
+  contour->first_edge = first_edge;
+
+  return contour;
+}
+
+void
+destroy_contour3d (contour3d *contour)
+{
+  g_free (contour);
+}
diff --git a/src/hid/gtk/contour3d.h b/src/hid/gtk/contour3d.h
new file mode 100644
index 0000000..8ce8d8a
--- /dev/null
+++ b/src/hid/gtk/contour3d.h
@@ -0,0 +1,9 @@
+typedef struct {
+  edge_ref first_edge;
+
+  /* STEP crap - to hell with encapsulation */
+  int face_bound_identifier;
+} contour3d;
+
+contour3d *make_contour3d (edge_ref first_edge);
+void destroy_contour3d (contour3d *contour);
